cmd/bedFormat/bedFormat.go

```diff
@@ -99,7 +133,64 @@ func bedFormat(s Settings) {
 		if s.ScaleNameFloat != 1 {
 			v.Name = fmt.Sprintf("%.8g", s.ScaleNameFloat*parse.StringToFloat64(v.Name))
 		}
-		bed.WriteBed(out, v)
+
+		if s.FdrAnnotation {
+			totalWindows++
+			if s.RawPValueAnnotationField >= len(v.Annotation) {
+				log.Fatalf("Error: rawPValueAnnotationField, %v, exceeds the length of the annotation slice in bed entry: %v.\n", s.RawPValueAnnotationField, len(v.Annotation))
+			}
+			if mapEntry, inMap = fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])]; inMap {
+				mapEntry.Count++
+				mapEntry.RawPValue = parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = mapEntry
+			} else {
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = FdrConverter{Count: 1, Rank: 0, RawPValue: parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField]), AdjPValue: 0}
+			}
+
+			bed.WriteBed(tmp, v)
+		} else {
+			bed.WriteBed(out, v)
+		}
+	}
+
+	if s.FdrAnnotation {
+		err = tmp.Close()
+		exception.PanicOnErr(err)
+
+		var currRank int = 0
+		var fdrSlice []FdrConverter = make([]FdrConverter, 0)
+		for i := range fdrMap {
+			fdrSlice = append(fdrSlice, fdrMap[i])
+		}
+
+		sort.Slice(fdrSlice, func(i, j int) bool { return compareFdrConverter(fdrSlice[i], fdrSlice[j]) == -1 })
+
+		for i := len(fdrSlice) - 1; i >= 0; i-- {
+			currRank += fdrSlice[i].Count
+			fdrSlice[i].Rank = currRank
```

Comment:
Note to self: 
- fdrSlice was sorted, by compareFdrConverter (	-1 means a.RawPValue < b.RawPValue, so rawValues are increasing aka ascending order after sorted, reference: https://pkg.go.dev/sort). 
- Then, attach currRank to fdrSlice entry in descending order, because we will start with the last entry, which is the largest value, and has rank 0 (and after that, smaller values, rank 1, 2, 3, etc.)
- take into account that currRank needs to include count of that entry
- then write out fdrSlice[i]'s true currRank